draft-irtf-cfrg-aead-limits.md

```diff
@@ -678,45 +697,209 @@ IA <= AEA
 {{mu-gcm-ae}} therefore contains the integrity limits.
 
 
-## AEAD_CHACHA20_POLY1305, AEAD_AES_128_CCM, and AEAD_AES_128_CCM_8
+## AEAD_CHACHA20_POLY1305
 
-There are currently no concrete multi-key bounds for AEAD_CHACHA20_POLY1305,
-AEAD_AES_128_CCM, or AEAD_AES_128_CCM_8. Thus, to account for the additional
-factor `u`, i.e., the number of keys, each `p` term in the confidentiality and
-integrity limits is replaced with `p / u`.
+Concrete multi-key bounds for AEAD_CHACHA20_POLY1305 are given in Theorem 7.8
+in {{ChaCha20Poly1305-MU}}, covering protocols with nonce randomization like
+TLS 1.3 {{TLS}} and QUIC {{?RFC9001}}.
 
-### AEAD_CHACHA20_POLY1305
+For this AEAD, n = 512, k = 256, t = 128, and r = 96; the length l is the sum
+of AAD and plaintext (in blocks of 128 bits).
 
-The combined confidentiality and integrity limit for AEAD_CHACHA20_POLY1305 is
-as follows.
+### Authenticated Encryption Security Limit {#mu-ccp-ae}
+
+<!--
+    From {{ChaCha20Poly1305-MU}} Theorem 7.8; for nonce randomization (XN transform).
+    
+    Let:
+        - d: the max. number of times any nonce is repeated across users
+        - \delta: the nonce-randomizer result's parameter
+        - d = r * (\delta + 1) - 1 < 2^9, \delta = 2 be fixed, satisfying Theorem 7.8
+        - this limits the number of encryption queries to q <= r * 2^(r-1) <= 2^101
+        - o, B <= 2^261 as required for Theorem 7.8
+    
+    We can simplify the Theorem 7.8 advantage bound as follows:
+        - 1st term:  v([constant]* l + 3)/2^t
+          Via Theorem 3.4, the more precise term is:  v * (2^25 * (l + 1) + 3) / 2^128
+          The 3v/2^t summand is dominated by the rest, so we simplify to
+            (v * (l + 1)) / 2^103
+        
+        - 2nd term:  d(o + q)/2^k
+          For d < 2^9 (as above) and o + q <= 2^145, this is dominated by the 1st term;
```

Comment:
These constraints are merely derived from comparing the 1st and 2nd terms, without further constraints.
How about the following explanation?

```suggestion
          For d < 2^9 (as above) and o + q <= 2^145, this is dominated by the 1st term;
          [[ 1st term <= 2nd term as long as v * (l + 1)/2^103 <= d(o + q)/2^256;
          i.e., o + q <= v * (l + 1) * 2^153 / d.
          Even for minimal values v = 1 and l = 1 in 1st term, with d < 2^9,
          this holds as long as o + q <= 2^145. ]]
```